plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/persistence/LibraryClasspathContainerSerializer.java

```diff
@@ -104,13 +107,22 @@ public void saveContainer(IJavaProject javaProject, LibraryClasspathContainer co
     if (stateFile == null) {
       logger.warning("Container state file cannot be created, save failed");
       return;
+    } else {
+      truncateFile(stateFile);
```

Comment:
The reason I added truncate was the following SO question:
http://stackoverflow.com/questions/15387567/does-fileoutputstream-truncate-an-existing-file

But by rereading it I realized it's not the same use-case we have here (we don't have competing threads writing and reading the file as these jobs use a scheduling rule obtained for the project so they should not try to touch the same file at the same time).

I'll remove the truncation.